x-pack/plugins/lens/public/visualizations/xy/add_layer.tsx

```diff
@@ -71,6 +90,36 @@ export function AddLayerButton({
     };
   };
 
+  const dataPanel = ({
+    type,
+    label,
+    icon,
+    disabled,
+    toolTipContent,
+  }: typeof supportedLayers[0]) => {
+    return {
+      panel: AddLayerPanelType.layerVisualizationType,
+      toolTipContent,
+      disabled,
+      name: <span className="lnsLayerAddButtonLabel">{label}</span>,
+      className: 'lnsLayerAddButton',
+      icon: icon && <EuiIcon size="m" type={icon} />,
+      ['data-test-subj']: `lnsLayerAddButton-${type}`,
+    };
+  };
+
+  const horizontalOnly = isHorizontalChart(state.layers);
+
+  const availableVisTypes = visualizationTypes.filter(
+    (t) => isHorizontalSeries(t.id as SeriesType) === horizontalOnly
+  );
+
+  // to decide: should we shift the availableVisTypes array?
+  const currentLayerVisType =
+    availableVisTypes.findIndex((t) => t.id === getDataLayers(state.layers)[0].seriesType) || 0;
```

Comment:
`getDataLayers` can return an empty array, even if at this point, it is probably very unlikely that, I think is better to be safe and check for that as you did in `x-pack/plugins/lens/public/visualizations/xy/visualization.tsx`

```suggestion
    availableVisTypes.findIndex((t) => t.id === getDataLayers(state.layers)?.[0]?.seriesType) || 0;
```